### DISCUSSION TITLE
SonOff-Basic R2 can't connect to WLAN after Update to 12.3.x

### DISCUSSION THREAD

**Discussion by baerenwaldfreund** [arendst/Tasmota#17471] [Q&A] [answered]
I tried to update several SonOff-Basic to firmware 12.3.x. But after the successfull update, the devices can not connect to WLAN:

`00:00:25.108 WIF: verbinden mit AP1 SSID_des_WLANS in Modus 11n als sonoff-05...
00:00:38.015 WIF: Verbindung fehlgeschlagen, da der AP nicht antwortet (timeout)`

After I go back to an older firmware 12.2.0 the device boots normally.

1st I've flashed the minimal.gz. Then the DE.gz-Firmware.

Anybody else has this problem?

**Comment by stickilino** [comment]
I have the same issue for Sonoff Basic R1. The Sonoff Basic R2 works well. Must be a bug in the firmware. 
After downgrade to 12.2 everything works fine again.

**Comment by Jason2866** [comment]
Without detailed logs there is no chance to help. None of all my upgraded devices have an issue.

**Reply by stickilino** [reply]
How do I get a log without access to the device?
Do you have a Basic R1 as well?

**Reply by Jason2866** [reply]
I have all Sonoff Basic variants. How have you managed to flash a older firmware without access? Dissasemble connect via Usb Serial adapter and you have logs via serial.

**Comment by arendst** [comment]
Pls sprovide logging like ``status 0`` and ``gpio all``

Also, do you connect external sensors to the Sonoff Serial connection? And if so what is connected.

**Comment by baerenwaldfreund** [comment]
OK, here are the logs via serial connection:
[FW-UPdate.log](https://github.com/arendst/Tasmota/files/10295858/FW-UPdate.log)

**Comment by baerenwaldfreund** [comment]
I've some new information about this bug.

After 12 min. the SonOff-Device connects to the WLAN. And the device works fine with FW 12.3.1. Then I unplugged it and tried again. But the device no longer connects to the WiFi.
[FW-UPdate2.log.txt](https://github.com/arendst/Tasmota/files/10295970/FW-UPdate2.log.txt)

**Comment by Jason2866** [comment]
Which Router do you have?
Edit: saw in log2, probably Fritzbox. 
Do you have mesh network or repeaters?

**Comment by baerenwaldfreund** [comment]
It's a Fritz!Box 7490 with the newest Fritz!OS 7.29. This Router ist part of a mesh (mesh-repeater).

EDIT:
I also have an 7590 as MeshMaster. Same problem.

**Reply by Jason2866** [reply]
Can you switch of the Fritz mesh devices and try if it connects than?
If yes your mesh setup is configured in a way Espressif Lwip stack can not handle.

**Comment by Jason2866** [comment]
If the connect problem is still there, we cant offer a solution. We dont know the real reason. 
There is a workaround implemented  `SetOption142 1` you can try.
Some Fritzbox users have problems (any Tasmota version). For most it does work without any issue.

**Comment by baerenwaldfreund** [comment]
I tried a connection to my lancom internet-router. The sonoff-device connects to the lancom1781vaw. I tried it only once.

I enabled the SetOption142 but this didn't help. After Update to the german 12.3.1 the device can't connect to the Fritz!Box-WLAN.

But it's no problem with the minimal-12.3.1. Then I tried the tasmota.bin.gz. (no-language-specific) After a few connection attempts it works also. I think it's an problem with the german firmware?!?!?

**Comment by Jason2866** [comment]
> I tried a connection to my lancom internet-router. The sonoff-device connects to the lancom1781vaw. I tried it only once.

No doubt since i have a LANCOM LN-830acn which works perfect :-)

Try latest development german firmware.

**Comment by baerenwaldfreund** [comment]
The german developer firmware 12.3.1.2 (build 20221223 15:17) didn't work.

EDIT:
build 20221223 18:17 didn't work too.

**Comment by stickilino** [comment]
I confirm from my side, using as well a fritzbox with repeater mesh and the german tasmota firmware. The two devices that fail have been in use for at least 3 years without any problems at all.

**Comment by Jason2866** [comment]
When using the english Tasmota firmware (develop and release) it does work?
Using the non working one with Fritzbox, do work with the Lancom router?

**Comment by Lassska** [comment]
I tasmotized two sOnOff Basic R2 modules and encountered the same phenomenon: After installing 12.3.1, I see the assigned IP as long as it is powered by the programmer and I can connect WIFI... After use the 220V power supply, WIFI no longer appears. If I return to 3.3V directly by programmer, I have wifi access again. I reverted to 12.2.0 until it is resolved!
I used tasmota.bin and two different routers with or without mesh (fritz and archer c2)

**Reply by Jason2866** [reply]
Can you measure the voltage on GPIO 0 when connected via the USB serial adapter to the PC?

**Reply by Lassska** [reply]
Hi
I just put the toys away. You only need to supply with 3.3V without rx and tx for wifi to be ok!
Honestly, I have no intention of doing an analysis that would lead, for example, to the need to cut a physical route or remove a part, since the previous version works with the module as it is manufactured...

**Reply by Jason2866** [reply]
@Lassska okay, so I just can help here. I have this devices and the do work with every version.

**Reply by Lassska** [reply]
ok Tomorrow we can compare exactly the boards and possibly the chips on the R2

**Reply by Lassska** [reply]
@Jason2866 u need any other info?

**Reply by Jason2866** [reply]
Yes, the measured voltage from GPIO 0 when connected to the USB serial adapter.
I do have a Sonoff Basic R2, it is a known borderline shitty device. Power Supply is to weak.
Thermal design and placement of esp chip and flash could have not been done worse.
It was the reason for doing the Tasmota-lite build. Reduce features to lower the MCU load.
It seems v12.3.x needs a little bit more power and the crappy Basic R2 fails.
Will investigate the next days, but I do not expect there is a solution. Bad hardware can’t be reliable fixed by software. Sonoff did very fast a new revision, since the had massive problems with the R2….
Best solution is to ditch the R2 in the bin and replace with a better alternative

**Reply by stickilino** [reply]
You might be right, but my R2s worked well since years and the software update causes trouble not the hardware.

**Reply by Jason2866** [reply]
New features, more MCU load more power needed -> R2 can’t handle and fails.
Usual with new software. Windows, MacOS…
For Sonoff R2 the train stops at 12.2.x

**Reply by stickilino** [reply]
But this means the automatic update procedure in OpenHab does not work, or need exceptions for the R2 to stay on 12.2 ...

**Reply by Jason2866** [reply]
@stickilino My known bad R2 does work with v12.3.1. But only when `sleep` is set at least to 20.
If this does not work and you want to use auto update on all devices, the R2 need’s replacing.

**Reply by Lassska** [reply]
@Jason2866 with the tasmota-sensors.bin 12.2.0 version that works ok, I now measured the voltage on GPIO0 fed with the programmer and there are values ​​between 3.15~3.30V
How about I check again?

**Reply by Jason2866** [reply]
@Lassska The values are okay, that’s why the device is working with the adapter. the values will probably not differ with an other firmware version. Device will work with v12.3.1 when using the usb serial adapter and sleep set to 20. So the issue (as always with the R2) is the device power supply.

**Comment by stickilino** [comment]
Hello Jason,
first, Happy New Year !!!
Thanks for your great support and engagement.

I just figured out: The trouble making Sonoff is not the Sonoff R1. It is indeed the Sonoff Basic R2 v1.0. The newer version: Sonoff RF R2 Power does not make any problems.

I will now try to figure out if it's only the German firmware making problems or as well the english version.
How do I get access to the log file once I cant connect to the sonoff via wifi?

Regards
Jörg

**Reply by Lassska** [reply]
Happy New Year!!! I have this problem with the English version for R2 V1

**Comment by Jason2866** [comment]
Getting closer, please try the tasmota-lite.bin version.
The R2 is known to be borderline.

**Reply by stickilino** [reply]
OK, let me check

**Reply by stickilino** [reply]
the lite version does not give a web-interface to connect 
I try to reconnect with the 12.2 version and make an OTA update

**Comment by stickilino** [comment, marked as answer]
OTA Upgrade to 12.3.1 minimal.bin works well, no problem.
OTA Upgrade to 12.3.1 tasmota.bin - Upload Failed - Not enough space
Tasmotizer Upgrade to 12.3.1 tasmota.bin - OK - but no Wifi connect possible afterwards
Tasmotizer Downgrade to 12.2 - everything work fine again

**Reply by Jason2866** [reply]
Please check if you have set `sleep 20`. Sleep is needed for the R2 !!

**Reply by Jason2866** [reply]
I tested with my R2, without sleep it fails, as it does with every Tasmota version when sleep is not enabled.
With `sleep 20` set it does work with v12.3.x too

See https://github.com/arendst/Tasmota/issues/17500#issuecomment-1370981742

**Comment by Jason2866** [comment]
Tasmota-lite not Tasmota
http://ota.tasmota.com/tasmota/release/tasmota-lite.bin.gz

**Reply by stickilino** [reply]
yes ... you send before I finish writing :-)

**Comment by stickilino** [comment]
OK,
it took me a long time to get it back to my network with 12.2
trying to flash tasmota-lite.bin via OTA says Upload Failed - Not enough space

**Reply by stickilino** [reply]
any command for console to free space?

**Reply by stickilino** [reply]
OK ... 
OTA upgrade to tasmota-minimal.bin works
then the update to tasmota-lite.bin works as well ... BUT --> with tasmota-lite same problem: No Wifi connect :-(

**Reply by stickilino** [reply]
enough for today !!!
12.2 works fine .. that it for now.

**Reply by Lassska** [reply]
Morgen morgen nur nicht heute sagen alle faule Leute :))))

**Comment by stickilino** [comment]
Hello Jason,
my Sonoff Basic R2 v1.0 is running stable on 12.2 with sleep 50.
When I change to sleep 20 I can't upgrade via OTA anymore.
Upgrading via Tasmotizer will reset the sleep value and I can't connect to the device.
Are you sure you have successfully tested this with a R2 v1.0 ?

**Reply by Jason2866** [reply]
Yup, since it is borderline, you may need higher sleep values. 50 is fine too.

**Reply by Jason2866** [reply]
As said, the R2 is one big piece of crap.

**Reply by stickilino** [reply]
I can't believe that this device is suddenly a big piece of crap ...
My Sonoff Basic R2 v1.0 work absolute stable since years and they will most likely work some more years with 12.2.x firmware.

Did you figure out which change has caused the trouble on the R2?
It seems that even the lite version does not work.
To my understanding this is a major issue with the latest firmware that needs to be resolved.
You could create a new Module for the Basic R2 v1.0 where the trouble making software parts are disabled or create a lite-firmware that works even with the R2...
Just saying: "R2 is a piece of crap" is not a solution!

**Reply by Jason2866** [reply]
You are invited to do. I wont burn my free time for this. Are you gonna pay for?
Tested latest version with my known flaky R2. Does work. See https://github.com/arendst/Tasmota/issues/17500#issuecomment-1370981742

**Reply by Lassska** [reply]
@stickilino It's called evolution! Nevertheless, when I have time I will look for some electrolytic capacitors larger in capacity but also in size to convince myself that the power supply from R2 does not provide a sufficiently stable voltage when the latest version is running

**Comment by HofeBY** [comment]
Hi,
ich hatte es auch, das tasmota32 nicht richtig compilierte. Tasmota dagegen ging.
Tasmota Version 12.3.1.2 von Theo arends hatte ich geladen. aber auch eine ältere 12 Version compilierte auch nicht.
Ich bin dann in "Visual Studio Code" auf das linke Würfelsymbol mit ausgeklinkter Ecke
dann auf "PlatformIO IDE" klicken und die Version von 2.5.5 auf 2.5.4 downgraden.
Dannach lief der Compilerlauf auch für Tasmota32 wieder und das Firmwareimage ließ sich auf
meinen "Sonoff Dual R3" flaschen.

hoffentlich hilft der Hinweis

Oberfränkische Grüße.

**Reply by Lassska** [reply]
@COtto1234567 danke
Ich grüße Sie auch aus Siebenbürgen

**Comment by baerenwaldfreund** [comment]
Ich habe jetzt mal alle Vorschläge zur Behebung des Problems ausprobiert. Es funktioniert keine. Die deutsche Version der aktuellen Firmware läuft wohl nicht mehr auf meinen SonOff Basic R1. Schade.

Sollte es wirklich ein Problem mit der Spannungsversorgung sein, wundert es mich, dass das Teil auch nicht bootet, wenn er noch am Programmer hängt und darüber versorgt wird.

Ich bleibe dann auf der letzten funktionierenden 12.2.0. Mal gucken, ob sich irgendwann der Fehler genauer analysieren lässt.
